config/module/versions.go

```diff
@@ -50,6 +54,11 @@ func newest(versions []string, constraint string) (string, error) {
 		return iv.GreaterThan(jv)
 	})
 
+	// Store whether the constraint contains a metadata requirement,
+	// so we can check if we should be returning a specific metadata version
+	constraintMeta := strings.SplitAfterN(cs.String(), "+", 2)
```

Comment:
In spite of the name `NewConstraint`, I believe this `cs` object can actually contain multiple constraints, if given an input string like `>= 1.0.0, < 2.0.0`. This `cs` object is a `Constraints`, which is a `[]*Constraint`, so I think we _should_ be able to iterate over it to check the individual constraints inside.

Of course, it's a little odd to combine an exact constraint with any other constraint... if you write something like `>=1.0.0, 1.2.0+foo` then there _is_ a reasonable result, but it's kinda an odd thing to ask Terraform to do. :thinking:

Reply:
I think then, we'd actually want to check (since only enforcing on explicit equality) that there is only one constraint present?